### ISSUE TITLE
notflix eating ram

### ISSUE THREAD

**Issue by Etaash-mathamsetty** [Bugswriter/notflix#19]
$ free -m

![image](https://user-images.githubusercontent.com/45927311/147865471-507fd2d9-ef0f-4c55-ae78-abc13f7c88f9.png)

this was after running notflix, hogged 3-5 gb of ram after running and closing it. I am still clueless on how to free the ram.

**Comment by PRESFIL** [comment]
`notflix` creates temporary directory in `/tmp` for WebTorrent. It can be deleted,
but it would be nice to make a **cli switch** that enables auto-deletion upon `notflix`
exits. Not all users have terabytes of RAM.

**Comment by Etaash-mathamsetty** [comment]
> `notflix` creates temporary directory in `/tmp` for WebTorrent. It can be deleted, but it would be nice to make a **cli switch** that enables auto-deletion upon `notflix` exits. Not all users have terabytes of RAM.

thanks, I can write some cleanup code at the end of the shellscript now

**Comment by Etaash-mathamsetty** [comment]
ok so adding `rm -dRf /tmp/webtorrent` at the end of the file is the fix :D

**Comment by phodina** [comment]
Well, better way would be to use trap to call this function on exit and place this call at the top of the script. I can write the MR
```
trap "rm -dRf /tmp/webtorrent" EXIT
```

**Comment by Etaash-mathamsetty** [comment]
> Well, better way would be to use trap to call this function on exit and place this call at the top of the script. I can write the MR
> 
> ```
> trap "rm -dRf /tmp/webtorrent" EXIT
> ```

yeah that's better, since if i ctrl-c then the files won't be delted

**Comment by PRESFIL** [comment]
Do you want under-downloaded files to hang in memory? Something more
reliable can be used:

```sh
trap "rm -dRf /temp/webtorrent" SIGINT SIGTERM ERR EXIT
```

(from [nyaa](https://github.com/yaemiku/nyaa))

In any case, it may be worth reconsidering the decision to store temporary files
in `/tmp/`. `nyaa` downloads everything to the `$HOME/Downloads/` directory.

**Comment by Monjurul-Hasan-Sohan** [comment]
Is it 
> /temp/webtorrent
or 
>/tmp/webtorrent

**Comment by Etaash-mathamsetty** [comment]
> Is it
> 
> > /temp/webtorrent
> > or
> > /tmp/webtorrent
99% sure it is for you
/tmp
